libraries/AP_Param/AP_Param.cpp

```diff
@@ -240,16 +238,13 @@ bool AP_Param::check_group_info(const struct AP_Param::GroupInfo *  group_info,
 }
 
 // check for duplicate key values
-bool AP_Param::duplicate_key(uint16_t vindex, uint16_t key)
+bool duplicate_key(std::unordered_map<uint16_t,uint16_t> &m,uint16_t key)
 {
-    for (uint16_t i=vindex+1; i<_num_vars; i++) {
-        uint16_t key2 = _var_info[i].key;
-        if (key2 == key) {
-            // no duplicate keys allowed
-            return true;
-        }
+    if(m.find(key)==m.end()){
```

Comment:
I don't think this is the correct key to be finding.

https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_Param/AP_Param.cpp#L159

That NO_SHIFT thing is a hack around the problem.  So it's the group_id which is important - needs to be compared against every other group-id generated (and probably other things).

We should probably create a patch which actually demonstrates the problem to make this simpler.

Apologies if I'm a bit vague here.  I don't have my head around parameter storage ATM :-)

Reply:
Have made the change to compare the group_ids.!Thanks